fastestimator/util/google_download_util.py

```
@@ -85,17 +148,18 @@ def download_file_from_google_drive(file_id: str, destination: str, max_retries:
         print(f"File {destination} already exists, skipping download.")
         return
 
-    for _ in range(max_retries):
+    for attempt in range(max_retries):
         if is_valid_file(destination):
             return
-        # Randomize a sleep interval before attempting to download in order to prevent multiple unit tests from hitting
-        # the drive simultaneously
-        time.sleep(random.randint(5, 10))
+        # Exponential backoff with jitter; base of 30s gives Google throttling windows time to clear
+        wait = (2**attempt) * random.uniform(30, 60)
```

Ruff (0.15.10)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)